File: src/main/java/com/example/interviewlandbackend/model/Content.java
package com.example.interviewlandbackend.model;


import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "contents")
public class Content implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String contentName;

    @OneToMany(
            mappedBy = "content" ,
            cascade = {
                    CascadeType.MERGE , CascadeType.PERSIST , CascadeType.REMOVE
            }
    )
    private List<Section> sectionList;


    public Content() {
    }

    public Content(String contentName) {
        this.contentName = contentName;
    }


    public int getId() {
        return id;
    }

    public String getContentName() {
        return contentName;
    }

    public List<Section> getSectionList() {
        return sectionList;
    }

    public void setContentName(String contentName) {
        this.contentName = contentName;
    }
}

File: src/main/java/com/example/interviewlandbackend/model/Content.kt
package com.example.interviewlandbackend.model

import java.io.Serializable
import javax.persistence.*


@Entity
@Table(name = "contents")
data class Content @JvmOverloads constructor(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = 0,

    var contentName: String,

    var description: String,

    @OneToMany(
        mappedBy = "content" ,
        cascade = arrayOf(CascadeType.ALL)
    )
    val sectionList: List<Section>? = ArrayList(),

    @OneToOne(cascade = arrayOf(CascadeType.ALL))
    @JoinColumn(name = "image_id")
    var image: Image,

    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: User?


):Serializable{





}
